### PR TITLE
Improve taproot nums point test

### PR DESCRIPTION
**Pull request by Eunovo**
Address @S3RK Reviews on https://github.com/S3RK/bip-352/pull/3
This is a work-in-progress

### CODE REVIEW

**Comment by S3RK** [comment]
LGTM. Do you want also to address scriptPubKey here on in the other PR?

You can take tweaked key calculation code from `generate_all_inputs_test` as I implemented it there

**Comment by Eunovo** [comment]
@S3RK  I used the `btcdeb` `tap` command to check my code in https://github.com/S3RK/bip-352/pull/5/commits/b01a4a9e856f3f9b67460924e47e526692362757 and the leaf hash that `tap` produces is different from what the code generates and I haven't been able to figure out why. See some console outputs below:
```
$ tap 50929b74c1a04954b78b4b6035e97a5e078a5a0f28ec96d547bfee9ace803ac0 1 0x205a1e61f898173040e20616d43e9f496fba90338a39faa1ed98fcbaeee4dd9be5ac
tap 5.0.24 -- type `tap -h` for help
WARNING: This is experimental software. Do not use this with real bitcoin, or you will most likely lose them all. You have been w a r n e d.
LOG: sign segwit taproot
Internal pubkey: 50929b74c1a04954b78b4b6035e97a5e078a5a0f28ec96d547bfee9ace803ac0
1 scripts:
- #0: 205a1e61f898173040e20616d43e9f496fba90338a39faa1ed98fcbaeee4dd9be5ac
Script #0 leaf hash = TapLeaf<<0xc0 || 205a1e61f898173040e20616d43e9f496fba90338a39faa1ed98fcbaeee4dd9be5ac>>
 → 9dc67742d203f7c919ffd84a1018e775ca50e46ae051ca0ede7f38d4aa1039fb
Tweak value = TapTweak(50929b74c1a04954b78b4b6035e97a5e078a5a0f28ec96d547bfee9ace803ac0 || 9dc67742d203f7c919ffd84a1018e775ca50e46ae051ca0ede7f38d4aa1039fb) = 79972ee6b22f373d54955c740a3e79d51bef741a90d5090ae9fdeea623ea38de
Tweaked pubkey = da6f0595ecb302bbe73e2f221f05ab10f336b06817d36fd28fc6691725ddaa85 (not even)
Resulting Bech32m address: bcrt1pmfhst90vkvpthee79u3p7pdtzrendvrgzlfkl550ce53wfwa42zsypfwpp
```
The `tap` commands produces a leaf hash of `9dc67742d203f7c919ffd84a1018e775ca50e46ae051ca0ede7f38d4aa1039fb` but the code produces `11185f79507f22c50baba3532d0c1cb7f169e320859fa5f21bae66a80de55e6a`

**Comment by josibake** [comment]
Took a quick look at the code and `get_leafp2tr_witness` seems incorrect: for the control block it is hashing the constant `b'leafhash'`. Not sure if this what's causing your specific issue, but definitely seems like it could be @Eunovo

**Comment by Eunovo** [comment]
> Took a quick look at the code and `get_leafp2tr_witness` seems incorrect: for the control block it is hashing the constant `b'leafhash'`. Not sure if this what's causing your specific issue, but definitely seems like it could be @Eunovo

`get_leafp2tr_witness` is not being used right now, so while it may be incorrect, something else has to be the cause of the problem

File: generate-test-vector.py [hunk 1]
```diff
@@ -1088,50 +1088,54 @@ def generate_taproot_with_nums_point_test():
...
+                }]
+                included_keys += [(key, True)]
+                included_pubkeys += [pub_key]
             else:
+                script = "20" + pub_key.get_bytes(True).hex() + "ac"
+                leaf_version = "c0"
+                annex = "50"
+                internal_key_bytes = internal_key.to_bytes(32, 'big')
+                leaf_hash = TaggedHash("TapLeaf", bytes.fromhex(leaf_version + script))
+                tap_tweak = TaggedHash("TapTweak", internal_key_bytes + leaf_hash)
```
Comment (S3RK) [inline generate-test-vector.py:1124]: You forgot to add size of script.

It should be `hash(v || compact_size(size of s) || s)`

I verified and the calculation is correct then, including the tweaked key

**Comment by S3RK** [comment]
> LGTM. Do you want also to address scriptPubKey here on in the other PR?
> 
> You can take tweaked key calculation code from `generate_all_inputs_test` as I implemented it there

I also realised that the code I was referring to was accidentally removed by 28c9c5a67fd108c1b8685220d0e285d99b5cd383

You can find more examples if you look up that commit in the console with `git show 28c9c5a67fd108c1b8685220d0e285d99b5cd383`

File: generate-test-vector.py (hunk 1 above)
Comment (josibake) [reply generate-test-vector.py:1124]: thanks @S3RK ! I feel kinda silly for not spotting this earlier :sweat_smile:

Comment (Eunovo) [reply generate-test-vector.py:1124]: @josibake @S3RK Resolved in https://github.com/S3RK/bip-352/pull/5/commits/4e5b3c637b712961b0ee8cd8c876ef661b8b70c8

**Comment by josibake** [comment]
LGTM, thanks @Eunovo !

**Pull request merged**